flyteidl/protos/flyteidl/core/types.proto

```diff
@@ -108,6 +108,14 @@ message UnionType {
     repeated LiteralType variants = 1;
 }
 
+// Defines a named tuple type
+// TODO: Add more docs
+message TupleType {
+    string tuple_name = 1;
+    repeated string order = 2;
+    map<string, LiteralType> fields = 3;
```

Comment:
Can we use repeated here, like

```proto
message TupleEntry {
  string name = 1;
  LiteralType literal_type = 2;
}
message TupleType {
    string tuple_name = 1;
    repeated TupleEntry = 2;
}
```

Reply:
> Other Considerations
An alternative approach to storing the literal values for NamedTuple and Tuple would be combining the order and the map into a list of messages, as shown below:

// !!! This is not the proposed implementation !!!
message FakeLiteralTupleField {
    string name = 1;
    Literal value = 2;
}

message FakeLiteralTupleMap {
    string tuple_name = 1;
    repeated FakeLiteralTupleField fields = 2;
}
While this structure may seem more straightforward, it has a drawback: looking up a field’s value by name requires iterating through the list, which results in O(n) complexity. In contrast, using a map allows for more efficient O(1) lookup times when retrieving the value of a field by its name. Although in practice the tuple size is usually small, the map-based approach is more scalable and efficient.